### PR TITLE
Expands per/unit mappings and updates the cache

### PR DESCRIPTION
**Pull request by zoobereq**
# What does this PR do ?

The current PR expands the list of "currency per unit" mappings contained in `data/money/per_unit.tsv` for EN TN. 

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Have you signed your commits? Use ``git commit -s`` to sign.
- [x] Do all unittests finish successfully before sending PR?
   1) ``pytest`` or (if your machine does not have GPU) ``pytest --cpu`` from the root folder (given you marked your test cases accordingly `@pytest.mark.run_only_on('CPU')`).
   2) Sparrowhawk tests ``bash tools/text_processing_deployment/export_grammars.sh --MODE=test ...``
- [x] If you are adding a new feature: Have you added test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
- [x] Have you added ``__init__.py`` for every folder and subfolder, including `data` folder which has .TSV files?
- [x] Have you followed codeQL results and removed unused variables and imports (report is at the bottom of the PR in github review box) ?
- [x] Have you added the correct license header `Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` to all newly added Python files?
- [x] If you copied [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
- [x] Remove import guards (`try import: ... except: ...`) if not already done.
- [x] If you added a new language or a new feature please update the [NeMo documentation](https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization/wfst/wfst_text_normalization.rst) (lives in different repo).
- [x] Have you added your language support to [tools/text_processing_deployment/pynini_export.py](tools/text_processing_deployment/pynini_export.py).
  

**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation
- [ ] Test

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

File: tests/nemo_text_processing/en/data_text_normalization/test_cases_address.txt [hunk 1]
```diff
@@ -7,3 +7,5 @@
 12 S 1st st~twelve South first Street
 1990 for the Ata ST~nineteen ninety for the Ata ST
 Main St.~Main St.
+Nancy lived at 1428 Elm St. It was a strange place.~Nancy lived at fourteen twenty eight Elm Street. It was a strange place.
+(incorrect normalization) Nancy lived at 1428 Elm St. 4 doors down lived her best friend.~(incorrect normalization) Nancy lived at fourteen twenty eight Elm Street four doors down lived her best friend.
```
Comment (tbartley94) [inline tests/nemo_text_processing/en/data_text_normalization/test_cases_address.txt:11]: Does the tester allow this formatting?

**Review by tbartley94 — APPROVED** [review]
LGTM (sorry didn't realize it was small changes)

File: tests/nemo_text_processing/en/data_text_normalization/test_cases_address.txt (hunk 1 above)
Comment (zoobereq) [reply tests/nemo_text_processing/en/data_text_normalization/test_cases_address.txt:11]: The tests (local and CI) pass without issues.

[Something similar](https://github.com/NVIDIA/NeMo-text-processing/blob/main/tests/nemo_text_processing/es/data_inverse_text_normalization/test_cases_ordinal.txt) was implemented for Spanish ITN.  @mgrafu suggested including that in lieu of documenting the edge case / limitation of the model.

Comment (tbartley94) [reply tests/nemo_text_processing/en/data_text_normalization/test_cases_address.txt:11]: Oh no, in the future you want to document edge cases. No one is going to check data files beyond us.
